Deintegrate CocoaPods

mokagio opened the pull request:
📘 Part of: https://linear.app/a8c/issue/AINFRA-700

Notice:

- how the `Package.resolved` file has now moved to the `xcodeproj` folder
- the source changes resulting from the new `google-cast` version, which I guess somehow did not get resolved at the time of https://github.com/Automattic/pocket-casts-ios/pull/3236

<!-- Please include a summary of what this PR is changing and why these changes are needed. -->

## To test

See green CI, where I also run a prototype build.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 500 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

danielebogo commented:
Thanks @mokagio ! I also opened a PR #3236 but I can close it and we can proceed with this one!

After merging your PR in the Google Cats repo, I created a new tag [1.0.1](https://github.com/Automattic/google-cast/tree/1.0.1)

mokagio committed:
Deintegrate CocoaPods

Notice how the `Package.resolved` file has now moved to the `xcodeproj`
folder and the source changes resulting from the new `google-cast`
version, which I guess somehow did not get resolved at the time of
https://github.com/Automattic/pocket-casts-ios/pull/3236

pocketcasts commented:
<!-- REUSE_ID: prototype-build-link -->

<details><summary><img align='top' src='https://raw.githubusercontent.com/buildkite/emojis/main/img-buildkite-64/firebase.png' width='20px' alt='App Icon' />📲 You can test the changes from this Pull Request in <b>Pocket Casts Prototype Build</b> by scanning the QR code below to install the corresponding build.</summary>
<table>
<tr>
  <td rowspan='6' width='260px'><img src='https://api.qrserver.com/v1/create-qr-code/?size=500x500&qzone=4&data=https%3A%2F%2Fappdistribution.firebase.google.com%2Ftesterapps%2F1%3A124902176124%3Aios%3A3999f43bfa1fcd1b1620f9%2Freleases%2F5r9v679bankh8%3Futm_source%3Dfastlane' width='250' height='250' /></td>
  <td><b>App Name</b></td><td>Pocket Casts Prototype Build</td>
</tr>
<tr><td><b>Build Number</b></td><td><code>10830</code></td></tr>
<tr><td><b>Version</b></td><td><code>PR #3241</code></td></tr>
<tr><td><b>Bundle ID</b></td><td><code>au.com.shiftyjelly.podcasts.prototype</code></td></tr>
<tr><td><b>Commit</b></td><td>ff5369d665e76718582a961e00483548647542f0</td></tr>
<tr><td><b>Installation URL</b></td><td><a href='https://appdistribution.firebase.google.com/testerapps/1:124902176124:ios:3999f43bfa1fcd1b1620f9/releases/5r9v679bankh8?utm_source=fastlane'>5r9v679bankh8</a></td></tr>
</table>
<em>Automatticians: You can use our internal self-serve MC tool to give yourself access to those builds if needed.</em>
</details>

mokagio commented on .gitignore:
```diff
@@ -1,4 +1,4 @@
-podcasts.xcodeproj/xcuserdata/
+**/xcuserdata/
```
Notice I haven't removed the Pods folder.

That's just a safety check to avoid it showing up in the local branches while migrating to this PR. I think it'll be safe to remove it in a couple of months, when all devs will have been on this version of the code for a while.

AliSoftware commented on README.md:
```diff
@@ -25,7 +25,7 @@ If you don't already have it, you need to install Bundler:
 `gem install bundler`
 
-Next you'll need to install all the dependencies needed for CocoaPods and FastLane using this script:
+Next you'll need to install all the dependencies needed for FastLane using this script:
```
```suggestion
Next you'll need to install all the dependencies needed for _fastlane_ using this script:
```
